homeassistant/components/switch/vesync.py

```diff
@@ -0,0 +1,101 @@
+"""
+Support for Etekcity Wifi Smart Switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.Vesync/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['vesync==1.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Etekcity Vesync Switch'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Set up Vesync switches."""
+    from vesync.api import VesyncApi
+
+    switches = []
+
+    try:
+        api = VesyncApi(config.get(CONF_USERNAME), config.get(CONF_PASSWORD))
+        _LOGGER.info("Connected to Vesync API")
+    except RuntimeError:
+        _LOGGER.error("Failed to connect to Vesync API")
```

Comment:
You should return otherwise the setup continues even if the API is not available.